pkg/cli/mirror/catalog_images.go

```diff
@@ -192,77 +199,95 @@ func addLayer(new, targetPath string) (v1.Layer, error) {
 	return layerFromFile(targetPath, new)
 }
 
-func buildCatalogLayer(ref, targetRef string, layer ...v1.Layer) error {
-	logrus.Debugf("Pulling image %s for processing", ref)
-	img, err := crane.Pull(ref)
+func (o *MirrorOptions) buildCatalogLayer(ctx context.Context, srcRef, targetRef, dir string, layers ...v1.Layer) error {
+
+	archs := []string{"amd64", "arm64", "ppc64le", "s390x"}
+	layoutPath := filepath.Join(dir, "layout")
+	if err := os.MkdirAll(layoutPath, os.ModePerm); err != nil {
+		return err
+	}
+
+	// Create an empty layout
+	p, err := layout.FromPath(layoutPath)
+	if err != nil {
```

Comment:
Is there a way to check if this is a non-existence error?

Reply:
Ended up removing the `FromPath` line and just did a `layout.Write`. The layout will always be empty anyway since we are using a temporary directory.